source/modulo_components/src/LifecycleComponent.cpp

```diff
@@ -243,7 +243,60 @@ bool LifecycleComponent::configure_outputs() {
+      auto message_pair = interface->get_message_pair();
+      switch (message_pair->get_type()) {
+        case MessageType::BOOL: {
+          auto publisher = this->create_publisher<std_msgs::msg::Bool>(topic_name, this->get_qos());
+          interface = std::make_shared<PublisherHandler<LifecyclePublisher<std_msgs::msg::Bool>, std_msgs::msg::Bool>>(
+                          PublisherType::LIFECYCLE_PUBLISHER, publisher)
+                          ->create_publisher_interface(message_pair);
+          break;
+        }
+        case MessageType::FLOAT64: {
+          auto publisher = this->create_publisher<std_msgs::msg::Float64>(topic_name, this->get_qos());
+          interface =
+              std::make_shared<PublisherHandler<LifecyclePublisher<std_msgs::msg::Float64>, std_msgs::msg::Float64>>(
+                  PublisherType::LIFECYCLE_PUBLISHER, publisher)
+                  ->create_publisher_interface(message_pair);
+          break;
+        }
+        case MessageType::FLOAT64_MULTI_ARRAY: {
+          auto publisher = this->create_publisher<std_msgs::msg::Float64MultiArray>(topic_name, this->get_qos());
+          interface = std::make_shared<PublisherHandler<
+              LifecyclePublisher<std_msgs::msg::Float64MultiArray>, std_msgs::msg::Float64MultiArray>>(
+                          PublisherType::LIFECYCLE_PUBLISHER, publisher)
+                          ->create_publisher_interface(message_pair);
+          break;
+        }
+        case MessageType::INT32: {
+          auto publisher = this->create_publisher<std_msgs::msg::Int32>(topic_name, this->get_qos());
+          interface =
+              std::make_shared<PublisherHandler<LifecyclePublisher<std_msgs::msg::Int32>, std_msgs::msg::Int32>>(
+                  PublisherType::LIFECYCLE_PUBLISHER, publisher)
+                  ->create_publisher_interface(message_pair);
+          break;
+        }
+        case MessageType::STRING: {
+          auto publisher = this->create_publisher<std_msgs::msg::String>(topic_name, this->get_qos());
+          interface =
+              std::make_shared<PublisherHandler<LifecyclePublisher<std_msgs::msg::String>, std_msgs::msg::String>>(
+                  PublisherType::LIFECYCLE_PUBLISHER, publisher)
+                  ->create_publisher_interface(message_pair);
+          break;
+        }
+        case MessageType::ENCODED_STATE: {
+          auto publisher = this->create_publisher<modulo_core::EncodedState>(topic_name, this->get_qos());
+          interface = std::make_shared<
+                          PublisherHandler<LifecyclePublisher<modulo_core::EncodedState>, modulo_core::EncodedState>>(
+                          PublisherType::LIFECYCLE_PUBLISHER, publisher)
+                          ->create_publisher_interface(message_pair);
+          break;
+        }
+        case MessageType::CUSTOM_MESSAGE: {
+          interface = this->output_configuration_callables_.at(name)(topic_name);
```

Comment:
do we want to call it custom_output_configuration_callables? It is getting rather long, but seems accuarte